src/openrct2/ride/Ride.cpp

```diff
@@ -397,8 +397,20 @@ money32 Ride::CalculateIncomePerHour() const
 
     if (currentShopItem != SHOP_ITEM_NONE)
     {
-        priceMinusCost += price[1];
-        priceMinusCost -= ShopItems[currentShopItem].Cost;
+        if (lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO)
+        {
+            uint32_t photosSold = no_secondary_items_sold;
+            int32_t rideTicketsSold = total_customers - no_secondary_items_sold;
+
+            // for now, ratio is just using all-time values of tickets sold and photos sold instead of last 5 minutes'
+            float_t photosRatio = rideTicketsSold > 0 ? ((float_t)photosSold / rideTicketsSold) : 0;
```

Comment:
We avoid floats in the main codebase. You can move the divide into the next calculation to keep it in the integer world.

Reply:
@duncanspumpkin You mean get rid of `photosRatio`?

Reply:
I'll have to cast it to a float at least. Is there any alternative to that?

Reply:
when you do the `(photosRatio * (price[1] - ShopItems[currentShopItem].Cost))` change it to `(photosSold * (price[1] - ShopItems[currentShopItem].Cost)) / rideTicketsSold`. You will need to have it so that rideTicketsSold is set as 1 if `rideTickestSold == 0`. That way you never need to make it a float.

Reply:
@duncanspumpkin I've done  a bit of refactoring to remove the float(basically what you've outlined). Imprecise, but it's okay I guess.